sql/role.go

```diff
@@ -0,0 +1,66 @@
+package sql
+
+import (
+  "context"
+  "database/sql"
+
+  "github.com/betr-io/terraform-provider-mssql/mssql/model"
+)
+
+func (c *Connector) GetRole(ctx context.Context, database, roleName string) (*model.Role, error) {
+  var role model.Role
+
+  err := c.
+    setDatabase(&database).
+    QueryRowContext(ctx,
+      "SELECT principal_id, name FROM [sys].[database_principals]  WHERE [type] = 'R' AND [name] = @roleName",
+      func(r *sql.Row) error {
+        return r.Scan(&role.RoleID, &role.RoleName)
+      },
+      sql.Named("roleName", roleName),
+    )
+  if err != nil {
+    if err == sql.ErrNoRows {
+      return nil, nil
+    }
+    return nil, err
+  }
+  return &role, nil
+}
+
+func (c *Connector) CreateRole(ctx context.Context, database, roleName string) error {
+  cmd := `DECLARE @sql nvarchar(max);
+  SET @sql = 'CREATE ROLE ' + QuoteName(@roleName);
```

Comment:
I think one potential improvement here could be to also support [authorization](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-role-transact-sql?view=sql-server-ver16#arguments). But that can probably be added in another PR:
`CREATE ROLE role_name [ AUTHORIZATION owner_name ]  `